File: library/src/main/java/cn/wandersnail/http/converter/StringResponseConverter.java
package cn.wandersnail.http.converter;

import androidx.annotation.NonNull;

import java.io.IOException;

import okhttp3.ResponseBody;
import retrofit2.Converter;

/**
 * 响应体为字符串
 * 
 * date: 2019/8/23 15:18
 * author: zengfansheng
 */
public class StringResponseConverter implements Converter<ResponseBody, String> {
    @Override
    public String convert(@NonNull ResponseBody value) throws IOException {
        return value.string();
    }
}

File: library/src/main/java/com/snail/network/converter/StringResponseConverter.kt
package com.snail.network.converter

import com.snail.network.exception.ConvertException
import okhttp3.ResponseBody

/**
 * 响应体的字符串
 *
 * date: 2019/2/27 23:49
 * author: zengfansheng
 */
class StringResponseConverter : ResponseConverter<String> {
    override fun convert(value: ResponseBody?): String? {
        if (value == null) {
            throw ConvertException("ResponseBody is null")
        } else {
            return value.string()
        }
    }
}